src/application-base/ApplicationBase.jsx

```diff
@@ -77,10 +44,18 @@ const propTypes = {
 };
 
 const ApplicationBase = ({
-  locale, customTranslatedMessages, translationsLoadingPlaceholder, themeName, fitToParentIsDisabled, children, unloadPromptIsDisabled,
+  locale, themeName, children, unloadPromptIsDisabled,
 }) => {
   const registeredPromptsRef = useRef();
 
+  const { localeOverride } = useTestOverrides(); // Allows us to test deployed applications in different locales.
+
+  const theme = useMemo(() => ({
+    // If the theme class name is undefined or an empty string, that indicates we have the root theme and should apply the root theme name.
+    name: themeName || rootThemeName,
+    className: themeName,
```

Comment:
Should this fallback to `themeConfig?.theme`?

Reply:
it could, but themeName is either null, or undefined, or it should be filled out here. We could be more explict but this does work. This logic hasn't changed from terra-application v1, it's just moved.